components/micronova.rst

```diff
@@ -0,0 +1,201 @@
+level shifting between 5v and 3V3.
+
+.. figure:: images/micronova_optocouplers.png
+    :align: center
+    :width: 100.0%
+
+    Optocoupler interface circuit (credit: philibertc)
+
+
+
+See the references below for all the details about te circuit.
+
+You can use the 5V output from the stove to power the ESP module, but you will have to put a voltage regulator in between to
+get 3v3.
+
+MicroNova configuration
+-----------------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      tx_pin: 5
+      rx_pin: 4
+      baud_rate: 1200
+      stop_bits: 2
+
+    micronova:
+      enable_rx_pin: 4
+      update_interval: 20s
+
+    text_sensor:
+      - platform: micronova
+        stove_state:
+          name: Stove status
+
+    number:
+      - platform: micronova
+        thermostat_temperature:
+          name: Thermostat temperature
+          step: 1
+      - platform: micronova
+        power_level:
+          name: Thermostat temperature
+
+    sensor:
+      - platform: micronova
+        room_temperature:
+          name: Room temperature
+        fumes_temperature:
+          name: Fumes temperature
+        water_temperature:
+          name: Water temperature
+          # every sensor can specify memory parameters
+          memory_location: 0x00
+          memory_address: 0x3B
+        water_pressure:
+          name: Water pressure
+        stove_power:
+          name: Stove power level
+        fan_speed:
+          fan_rpm_offset: 240
+          name: Fan RPM
+        memory_address_sensor:
+          memory_location: 0x20
+          memory_address: 0x7d
+          name: Custom Address sensor
+
+    switch:
+      - platform: micronova
+        stove:
+          name: Stove on/off switch
+
+    button:
+      - platform: micronova
+        custom_button:
+          name: Custom button
+          memory_location: 0x20
+          memory_address: 0x7d
+          memory_data: 0x08
+
+
+Micronova variables:
+~~~~~~~~~~~~~~~~~~~~
+- **enable_rx_pin** (**Required**, :ref:`config-pin`): Output pin to be used to switch the line between RX en TX.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval that the sensors should be checked.
+  Defaults to 60 seconds.
+
+.. note::
+
+    For most Micronova boards the default **memory_location** and **memory_address** parameters will work so you should
+    not specify them. However your Micronova boad may require you to specify alternate values. So every (text)sensor, button,
+    switch or number accepts these parameters:
+
+    - **memory_location** (*Optional*): The memory location where the parameter must be read. For most stoves this is 0x00 for RAM
+      or 0x20 for EPROM.
+    - **memory_address** (*Optional*): The address where the parameter is stored.
+
+
+Text Sensor variables:
+~~~~~~~~~~~~~~~~~~~~~~
+- **stove_state** (*Optional*): The current stove state.
+- All other options from :ref:`Text Sensor <config-text_sensor>`.
```

Comment:
Please alter all of the sections for each platform to be like this

```suggestion
Text Sensors
------------

(Example here)

Configuration variables:
~~~~~~~~~~~~~~~~~~~~~~~~

- **stove_state** (*Optional*): The current stove state.
- All other options from :ref:`Text Sensor <config-text_sensor>`.
```

Reply:
Done, but maybe not perfect to your taste. Feel fee to steer me in the right direction.